pinot-query-planner/src/main/java/org/apache/pinot/query/planner/plannode/MailboxSendNode.java

```diff
@@ -50,8 +53,42 @@ public MailboxSendNode(int stageId, DataSchema dataSchema, List<PlanNode> inputs
     _sort = sort;
   }
 
+  public MailboxSendNode(int stageId, DataSchema dataSchema, List<PlanNode> inputs,
+      int receiverStage, PinotRelExchangeType exchangeType,
+      RelDistribution.Type distributionType, @Nullable List<Integer> keys, boolean prePartitioned,
+      @Nullable List<RelFieldCollation> collations, boolean sort) {
+    this(stageId, dataSchema, inputs, toBitSet(receiverStage), exchangeType, distributionType, keys, prePartitioned,
+        collations, sort);
+  }
+
+  private static BitSet toBitSet(int receiverStage) {
+    BitSet bitSet = new BitSet(receiverStage + 1);
+    bitSet.set(receiverStage);
+    return bitSet;
+  }
+
+  public MailboxSendNode(int stageId, DataSchema dataSchema, List<PlanNode> inputs,
+      PinotRelExchangeType exchangeType, RelDistribution.Type distributionType, @Nullable List<Integer> keys,
+      boolean prePartitioned, @Nullable List<RelFieldCollation> collations, boolean sort) {
+    this(stageId, dataSchema, inputs, null, exchangeType, distributionType, keys, prePartitioned, collations, sort);
+  }
+
+  public BitSet getReceiverStages() {
+    Preconditions.checkState(!_receiverStages.isEmpty(), "Receivers not set");
+    return (BitSet) _receiverStages.clone();
+  }
```

Comment:
nit: small doc comment clarifying that modifying the returned bitset doesn't affect this node's receiver stages might be useful.

Reply:
I'm not a fan of leaking these constraints, even if we document it.

In the next PR (spools actually used in queries) I changed this code to return a `clone()`. I can apply this change in this PR as well.

Reply:
Oh! I misread you. I though the code wasn't returning a copy and you wanted to say that the returned BitSet should not be modified.

About your actual comment: I don't think that needs to be commented. Specifically, I think that should be the default. Even if we don't do defensive copies, by default we should never modify structures returned by objects. It is a bit like annotating with `@NotNull`. In the same way we assume that everything is not nullable by default, we should assume we should not change mutable properties by default.

Reply:
In my next push I'll remove this method and substitute it with a more specific one that will be easier to use and won't be affected by this undefined external mutability problem